torch/__init__.py

```diff
@@ -2077,3 +2077,43 @@ def _constrain_as_size(symbol, min: Optional[builtins.int] = None, max: Optional
 
 from . import _logging
 _logging._init_logs()
+
+
+def _import_device_backends():
+    """
+    Leverage the Python plugin mechanism to load out-of-the-tree device extensions.
+    See this RFC: https://github.com/pytorch/pytorch/issues/122468
+    """
+    if sys.version_info < (3, 10):
+        from importlib_metadata import entry_points
+    else:
+        from importlib.metadata import entry_points
+
+    for backend_extension in entry_points(group='torch.backends'):
+        try:
+            # Just load the extension without calling
+            backend_extension.load()
+        except Exception as err:
+            print(f"Failed to load the backend extension: {backend_extension.name}")
```

Comment:
Why silence this failure rather than letting it stop the user code? 
```suggestion
            raise RuntimeError(f"Failed to load the backend extension: {backend_extension.name}. You can disable extension auto-loading with TORCH_DEVICE_BACKEND_AUTOLOAD=0.") from err
```